module/optee/mbx/test/mod_optee_mbx_unit_test.c

```diff
@@ -27,10 +30,313 @@ void tearDown(void)
 {
 }
 
+struct mbx_device_ctx fake_optee_mbx_ctx[FAKE_SCMI_SERVICE_IDX_COUNT];
+
+/*!
+ * \brief mbx unit test: dummy test.
+ *
+ *  \details TBF
+ */
+void test_optee_mbx_init_test(void)
+{
+    int status;
+    fwk_id_t module_id;
+
+    module_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+
+    /* Element count can't be null */
+    status = mbx_init(module_id, 0, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* Fail to allocate module cxt */
+    fwk_mm_calloc_ExpectAndReturn(
+        1,
+        sizeof(struct mbx_device_ctx),
+        NULL);
+    status = mbx_init(module_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_NOMEM, status);
+
+    /* Allocate module context */
+    fwk_mm_calloc_ExpectAndReturn(
+        1,
+        sizeof(struct mbx_device_ctx),
+        fake_optee_mbx_ctx);
+    status = mbx_init(module_id, FAKE_SCMI_SERVICE_IDX_COUNT, NULL);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+void test_optee_mbx_element_init_test(void)
+{
+    int status;
+    fwk_id_t device_id;
+    struct mod_optee_mbx_channel_config *device_config;
+
+    /* Set fake config data */
+    device_config =
+        (struct mod_optee_mbx_channel_config *)mbx_element_table[FAKE_SCMI_SERVICE_IDX_OSPM_0]
+            .data;
+
+    /* Successfully init element */
+    device_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(device_id, 0);
+    mutex_init_ExpectAnyArgs();
+    status = mbx_device_init(device_id, 0, device_config);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+void test_optee_mbx_bind_test_r0(void)
+{
+    int status;
+    fwk_id_t module_id;
+
+    /* Bind module always succeed because thre is nothing to do ... */
+    module_id = FWK_ID_MODULE(FWK_MODULE_IDX_OPTEE_MBX);
+    fwk_id_is_type_ExpectAndReturn(module_id, FWK_ID_TYPE_MODULE, true);
+    status = mbx_bind(module_id, 0);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    fwk_id_is_type_ExpectAndReturn(module_id, FWK_ID_TYPE_MODULE, true);
+    status = mbx_bind(module_id, 1);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    fwk_id_is_type_ExpectAndReturn(module_id, FWK_ID_TYPE_MODULE, true);
+    status = mbx_bind(module_id, 2);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+
+    /* Then bind something else at any round except round 1 always succeed ! */
+    module_id = FWK_ID_SUB_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_SERVICE_IDX_OSPM_0, 0);
+    fwk_id_is_type_ExpectAndReturn(module_id, FWK_ID_TYPE_MODULE, false);
+    status = mbx_bind(module_id, 0);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+
+    module_id = FWK_ID_SUB_ELEMENT(FWK_MODULE_IDX_OPTEE_MBX, FAKE_SCMI_SERVICE_IDX_OSPM_0, 0);
+    fwk_id_is_type_ExpectAndReturn(module_id, FWK_ID_TYPE_MODULE, false);
+    status = mbx_bind(module_id, 2);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+void test_optee_mbx_bind_request_test(void)
```

Comment:
I don't have strong opinion. just didn't want to duplicate too much because I usually reuse same variable so i was afraid to generate lot of code line for few effective test

Reply:
I am not enforcing as well 😄 I am just in favour of writing simple small unit tests with indicative names.
Let's see other folks' opinions on this.